### PR TITLE
DON-880: Don't display terms for apple pay or google pay, adjust matomo event for wallet payments

### PR DESCRIPTION
**Pull request by bdsl**
We display our own terms, so these should not be necassary

### CODE REVIEW

File: src/app/donation-start/donation-start-form/donation-start-form.component.ts [hunk 1]
```diff
@@ -1840,28 +1844,13 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
...
-    }
 
-    console.log(stripe_donation_method);
+    const stripeMethod = stripe_donation_method || 'undefined';
 
     this.matomoTracker.trackEvent(
       'donate',
-      action,
-      `Stripe Intent processing or done for donation ${donation.donationId} to campaign ${donation.projectId}`,
+      'stripe_card_payment_success',
```
Comment (NoelLH) [inline src/app/donation-start/donation-start-form/donation-start-form.component.ts:1852]: @bdsl I was thinking we would still vary this to `stripe_prb_payment_success` for easier reporting of headline stats (completion numbers on wallets), while also using the below addition to help "zoom in" on any reported issues.

File: src/app/donation-start/donation-start-form/donation-start-form.component.ts [hunk 2]
```diff
@@ -630,8 +630,7 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
    * According to stripe docs https://stripe.com/docs/js/element/events/on_change?type=paymentElement the change event has
```
Comment (bdsl) [inline src/app/donation-start/donation-start-form/donation-start-form.component.ts:630]: also I'd be happier if we didn't have to listen to changes in the payment element and instead this data came as part of the `Promise<PaymentIntentResult>` returned when we call `Stripe.confirmPayment`. But `PaymentIntentResult` just consists of the intent or an error.

Comment (bdsl) [reply src/app/donation-start/donation-start-form/donation-start-form.component.ts:630]: Right - done in 86713a52

File: src/app/donation-start/donation-start-form/donation-start-form.component.ts (hunk 1 above)
Comment (bdsl) [reply src/app/donation-start/donation-start-form/donation-start-form.component.ts]: Right - done in https://github.com/thebiggive/donate-frontend/commit/86713a524bb3ec947dfaf9657a9c9f95db3bf6d5

File: src/app/donation-start/donation-start-form/donation-start-form.component.ts [hunk 3]
```diff
@@ -1840,28 +1844,17 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
...
-                action = 'stripe_card_payment_success';
-            }
-            action = 'unknown_method_'+ stripe_donation_method+'_payment_success';
-            break;
-    }
 
-    console.log(stripe_donation_method);
+    const stripeMethod = stripe_donation_method || 'undefined';
+
+    // if not card then we assume it must be prb i.e. Google Pay or Apple Pay.
```
Comment (bdsl) [inline src/app/donation-start/donation-start-form/donation-start-form.component.ts:1850]:
```suggestion
    // if not card then we assume it must be a Payment Request Button i.e. Google Pay or Apple Pay.
```

**Review by NoelLH — APPROVED** [review]

**Pull request merged**